### PR TITLE
[3313] Avoid sending domain events for no-op updates

### LINKED ISSUE #3313: Avoid sending domain events for no-op "updates"
Currently operations like `org.eclipse.sirius.web.domain.boundedcontexts.project.services.ProjectUpdateService.renameProject(UUID, String)` *always* send a `ProjectNameUpdatedEvent` even if the "new" name is actually the same as before. This could be avoided so that listeners only get events for actual changes.

### PR DESCRIPTION
**Pull request by pcdavid**
Bug: https://github.com/eclipse-sirius/sirius-web/issues/3313

### CODE REVIEW

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/project/services/ProjectUpdateService.java [hunk 1]
```diff
@@ -56,14 +56,13 @@ public IResult<Void> renameProject(UUID projectId, String newName) {
             result = new Failure<>(this.messageService.notFound());
         } else {
             var project = optionalProject.get();
-            project.updateName(newName);
-
-            this.projectRepository.save(project);
+            if (!Objects.equals(project.getName(), newName)) {
+                project.updateName(newName);
+                this.projectRepository.save(project);
+            }
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/project/services/ProjectUpdateService.java]: Testing for this should be done in `Project#updateName`

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/RepresentationDataUpdateService.java [hunk 2]
```diff
@@ -47,8 +47,10 @@ public IResult<Void> updateContent(UUID id, String content) {
         var optionalRepresentationData = this.representationDataRepository.findById(id);
         if (optionalRepresentationData.isPresent()) {
             var representationData = optionalRepresentationData.get();
-            representationData.updateContent(content);
-            this.representationDataRepository.save(representationData);
+            if (!representationData.getContent().equals(content)) {
+                representationData.updateContent(content);
+                this.representationDataRepository.save(representationData);
+            }
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/RepresentationDataUpdateService.java]: This should be in RepresentationData#updateContent

**Review by sbegaudeau — CHANGES_REQUESTED** [review]

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/project/services/ProjectUpdateService.java (hunk 1 above)
Comment (pcdavid) [reply packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/project/services/ProjectUpdateService.java]: Done.

File: packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/RepresentationDataUpdateService.java (hunk 2 above)
Comment (pcdavid) [reply packages/sirius-web/backend/sirius-web-domain/src/main/java/org/eclipse/sirius/web/domain/boundedcontexts/representationdata/services/RepresentationDataUpdateService.java]: Done.

File: CHANGELOG.adoc [hunk 3]
```diff
@@ -80,7 +80,9 @@ They still support returning an `java.time.Instant` object directly.
 - https://github.com/eclipse-sirius/sirius-web/issues/3309[#3309] [diagram] Hide header border-bottom when the node is collapsed
 - https://github.com/eclipse-sirius/sirius-web/issues/3285[#3285] Make the icons in the diagram palette use the default color of Sirius Web theme
 - https://github.com/eclipse-sirius/sirius-web/issues/3312[#3312] Make the details view more compact
-
+- https://github.com/eclipse-sirius/sirius-web/issues/3285[#3285] Makes the icons in the diagram palette use the default color of Sirius Web theme
+- https://github.com/eclipse-sirius/sirius-web/issues/3285[#3285] Makes the icons in the diagram palette use the default color of Sirius Web theme
```
Comment (sbegaudeau) [inline CHANGELOG.adoc]: Small issue with the rebase, I'll fix that while rebasing the PR

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/domain/DomainEventstTest.java [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Integration tests for domain events publication.
+ *
+ * @author pcdavid
+ */
+@Transactional
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class DomainEventstTest  extends AbstractIntegrationTests {
```
Comment (sbegaudeau) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/domain/DomainEventstTest.java]: I'll fix the small typo here and the double space too.

**Review by sbegaudeau — APPROVED** [review]

**Pull request merged**